pkg/asset/agent/gencrypto/authconfig.go

```diff
@@ -0,0 +1,99 @@
+package gencrypto
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+
+	"github.com/openshift/installer/pkg/asset"
+)
+
+// AuthConfig is an asset that generates ECDSA public and private keys.
+type AuthConfig struct {
+	PublicKey, PrivateKey string
+}
+
+var _ asset.WritableAsset = (*AuthConfig)(nil)
+
+// Dependencies returns the assets on which the AuthConfig asset depends.
+func (a *AuthConfig) Dependencies() []asset.Asset {
+	return []asset.Asset{}
+}
+
+// Generate generates the auth config for agent installer APIs.
+func (a *AuthConfig) Generate(dependencies asset.Parents) error {
+	PublicKey, PrivateKey, err := keyPairPEM()
+	if err != nil {
+		return err
+	}
+	a.PublicKey = PublicKey
+	a.PrivateKey = PrivateKey
+
+	return nil
+}
+
+// Name returns the human-friendly name of the asset.
+func (a *AuthConfig) Name() string {
+	return "Agent Installer API Auth Config"
+}
+
+// Load returns the auth config from disk.
+func (a *AuthConfig) Load(f asset.FileFetcher) (bool, error) {
+	// The AuthConfig will not be needed by another asset so load is noop.
+	// This is implemented because it is required by WritableAsset
+	return false, nil
+}
+
+// Files returns the files generated by the asset.
+func (a *AuthConfig) Files() []*asset.File {
+	// Return empty array because File will never be loaded.
+	return []*asset.File{}
+}
+
+// Reused from assisted-service.
+// https://github.com/openshift/assisted-service/blob/d3c0122452c74ad208055b8b6ee412812431a83f/internal/gencrypto/keys.go#L13-L54
```

Comment:
Looks fine overall. Although generating the keys doesn't appear too complicated, I wonder if it would be worth asking assisted (@carbonin) to make this function public in the event there are security fixes or implementation changes later on.

Reply:
@carbonin WDYT about making the functionality public in assisted so that the installer code can import the functionality from assisted-service?

Reply:
Sure, make a PR and ping me in it.